### ISSUE TITLE
VPAT -- Radio Button grouping

### ISSUE THREAD

**Issue by pbailie** [Submitty/Submitty#2896]
Accessibility readers interpret `<fieldset>` and `<legend>` to identify and describe radio button and checkbox groups.  However, `<fieldset>` and `<legend>` seem to be used for strictly visual grouping that is inconsistent with accessibility tools.  

Minimally, the use of `<fieldset>` and `<legend>` needs to better correspond with accessibility tools to group radio buttons and checkboxes (and perhaps any other pertinently related controls corresponding to the radio buttons / checkboxes).  Optimally, HTML should not be used for visual aesthetics.  That is the purview of CSS.

This issue is particularly prevalent among the Gradeable creation and editing forms.

The following is advice provided by WAVE:

- `<fieldset>` does not exist for a radio button group (WAVE  _alert_):

> **Missing fieldset**
> **What It Means**
> A group of check boxes or radio buttons is not enclosed in a fieldset.
> **Why It Matters**
> A fieldset provides a visual and structural grouping of related form elements. It is typically necessary for groups of check boxes or radio buttons where a higher level description (called a legend) is necessary to understand the function of the check boxes or radio buttons. The description will be identified by a screen reader only if provided in a fieldset legend.
> **How to Fix It**
> Determine whether the grouping of check boxes or radio buttons has or needs text that explains the purpose of the check boxes or radio button grouping. If so, mark up the group within a fieldset and put the group description in a legend element.
> **The Algorithm... in English**
> Two or more checkbox or radio input elements within a form have the same name value, but are not enclosed in a fieldset.
> **Standards and Guidelines**
> [Section 508 (n)](http://webaim.org/standards/508/checklist#standardn)
> [1.1.1 Non-text Content (Level A)](http://webaim.org/standards/wcag/checklist#sc1.1.1)
> [1.3.1 Info and Relationships (Level A)](http://webaim.org/standards/wcag/checklist#sc1.3.1)
> [2.4.6 Headings and Labels (Level AA)](http://webaim.org/standards/wcag/checklist#sc2.4.6)
> [3.3.2 Labels or Instructions (Level A)](http://webaim.org/standards/wcag/checklist#sc3.3.2)

-  `<fieldset>` exists but `<legend>` does not (WAVE _alert_):

> **Fieldset missing legend**
> **What It Means**
> A fieldset does not have a legend.
> **Why It Matters**
> A fieldset legend presents a description of the form elements within a fieldset and is especially useful to screen reader users. A legend should be provided when a higher level description is necessary for groups of check boxes, radio buttons, or other form controls.
> **How to Fix It**
> If a higher level description is necessary for the user to understand the function or purpose of the controls within the fieldset, provide this description within the <legend>. If this description or grouping is not necessary, the fieldset should probably be removed. Note that the legend is repeated to screen reader users for each form control within the fieldset.
> **The Algorithm... in English**
> A fieldset does not have a legend or the legend is empty.
> **Standards and Guidelines**
> [Section 508 (n)](http://webaim.org/standards/508/checklist#standardn)
> [1.1.1 Non-text Content (Level A)](http://webaim.org/standards/wcag/checklist#sc1.1.1)
> [1.3.1 Info and Relationships (Level A)](http://webaim.org/standards/wcag/checklist#sc1.3.1)
> [2.4.6 Headings and Labels (Level AA)](http://webaim.org/standards/wcag/checklist#sc2.4.6)
> [3.3.2 Labels or Instructions (Level A)](http://webaim.org/standards/wcag/checklist#sc3.3.2)

**Comment by bmcutler** [comment]
@mktung I'm going to merge #4041, but please address @pbailie 's comment about one last legend in a future PR.
